using-timescaledb/compression.md

```diff
@@ -295,14 +295,80 @@ SELECT compress_chunk(i) from show_chunks('conditions', newer_than, older_than)
+command in much the same way you can manually compress an individual chunk:
 
-TimescaleDB does not support inserts or updates into a compressed chunk. To insert
-or update data, we must first decompress the target chunks. But before we can do
-that, we need to turn off our compression policy. Otherwise that policy will attempt
-to re-compress the chunks that we are currently working on (not the desired result).
-To accomplish this we first find the job_id of the policy using:
+``` sql
+SELECT decompress_chunk('_timescaledb_internal._hyper_2_2_chunk');
+```
+
+Similar to above, you can also decompress a set of chunks based on a
+time range by first looking up this set of chunks via `show_chunks`:
+
+``` sql
+SELECT decompress_chunk(i) from show_chunks('conditions', newer_than, older_than) i;
+```
+
+Decompression might often be employed in the event that you need to backfill or
+update data that lives in a compressed chunk, as TimescaleDB does not currenty
+support modifying (inserting into, updating, deleted from) compressed chunks.
+
+Next we walk you through the instructions for preparing your table for
+inserting or backfilling data.   The general approach has four steps:
+
+1. Temporarily turn off any compression policy (as otherwise that policy will attempt
+to re-compress the chunks that we are currently working on)
+
+1. Decompress chunks that will be effected by modifications or backfill
+
+1. Perform the modifications or backfill
+
+1. Re-enable compression policy (which will have the effect of recompressing
+any of our recently-decompressed chunks)
+
+In the next sections, we describe some automated helper functions we provide
+that perform all five steps in a more automatic fashion, and then also walk you
+through more manual instructions.
+
+#### Automatically decompressing chunks for backfill
+
+In the [TimescaleDB Extras][timescaledb-extras] github repository, we provide
+explicit functions for [backfilling batch data to compressed
+chunks][timescaledb-extras-backfill], which is useful for inserting a *batch*
+of backfilled data (as opposed to individual row inserts). By "backfill", we
+mean inserting data corresponding to a timestamp well in the past, which given
+its timestamp, already corresponds to a compressed chunk.
+
+To use this procedure:
+
+- First create a temporarily table with the same schema as the hypertable (in this example, `cpu`) that we are backfilling into:
+
+```sql
+CREATE TEMPORARY TABLE cpu_temp AS SELECT * FROM cpu WITH NO DATA;
+```
+
+- Second, insert data into the temporarily table.
+
+- Third, use a supplied backfill procedure to perform the above steps: halt
+  compression policy, identify those compressed chunks to which the backfilled
+  data corresponds, decompress those chunks, insert data from the temporarily
+  table into the main hypertable, and then re-enable compression policy:
+
+```sql
+CALL decompress_backfill(staging_table=>'cpu_temp'::regclass,
```

Comment:
Does this function really require parameter casting? I would assume that both parameters are of `regclass` type and in that case one could elide casts.

Reply:
cc: @davidkohn88